### ISSUE TITLE
Getting 404 response for artifacts if I specify tag.

### ISSUE THREAD

**Issue by schmidtw** [ncipollo/release-action#43]
**Release succeeds, but released files are not available/404**

It appears that when I add the tag field like how I think you have it documented, the files are "uploaded" and everything looks amazing, but when I try to download them they aren't really there.

**To Reproduce**
This is the configuration I used that caused the problem:
```
      - name: create release
        id: create_release
        uses: ncipollo/release-action@v1
        with:
          tag: ${{ github.ref }}
          name: ${{ steps.prep.outputs.rname }}
          draft: false
          prerelease: false
          bodyFile: ${{ steps.prep.outputs.body }}
          artifacts: "artifacts/*"
          token: ${{ secrets.GITHUB_TOKEN }} 
```

When I remove the `tag` field, the files are uploaded & all goes to plan.

This works great:
```
      - name: create release
        id: create_release
        uses: ncipollo/release-action@v1
        with:
          name: ${{ steps.prep.outputs.rname }}
          draft: false
          prerelease: false
          bodyFile: ${{ steps.prep.outputs.body }}
          artifacts: "artifacts/*"
          token: ${{ secrets.GITHUB_TOKEN }} 
```

**Expected behavior**
Either the documentation needs to be updated or there is a bug in the tag handling code.

**Comment by ncipollo** [comment]
Do you have a link to the generated release which is 404'ing when you attempt to download the artifact? I'm actually surprised the github API isn't failing to create the release. If you take a look at the actual tag you are providing it's the whole ref (not the just the tag). For example:  https://github.com/schmidtw/curlws/runs/2445544223?check_suite_focus=true#step:6:3

If you take a look at what I am doing in the action, I'm actually using the same ref field, but I'm stripping off the `refs/tags` portion of it before submitting to the github API - https://github.com/ncipollo/release-action/blob/a43fb1aa8280333cf1a30fd41a77896244d764b8/src/Inputs.ts#L153

**Comment by schmidtw** [comment]
I'm glad you found schmidw/curlws as that was a test repo to get the release right.  I put it back in the error mode & ran it again.

Here is a release that exhibits this problem:
https://github.com/schmidtw/curlws/releases/tag/refs%2Ftags%2Fv0.0.27

I did see a few that were `404` went away, but I didn't keep careful records.

**Comment by ncipollo** [comment]
Interesting, I think you may be running into a github issue. If you look at the the URL of this release you'll notice that it contains the (encoded) ref path - https://github.com/schmidtw/curlws/releases/tag/refs%2Ftags%2Fv0.0.27. If you compare this against a normal release you'll see the difference - https://github.com/schmidtw/curlws/releases/tag/v0.0.26.

Another interesting side effect of the weird tag is this release doesn't seem to show up in your releases page -https://github.com/schmidtw/curlws/releases. 

Long story short, the tag you were originally providing was `refs/tags/<tag>`. This looks like it messes with the github API. Removing the `tag` input in this scenario is the correct thing to do (since the action will automatically use the ref put strip out the bit which messes with Github).

**Comment by schmidtw** [comment]
Cool.  Yeah I think you nailed it.  I wanted to both ask a question & post some details in case someone else made the same mistake.  I was coming from the older https://github.com/actions/create-release.  Here is an example of the previous use:
https://github.com/xmidt-org/petasos/blob/main/.github/workflows/release.yml#L30

Thanks!
